tests/dragonfly/replication_test.py

```diff
@@ -945,3 +945,51 @@ async def test_replication_info(df_local_factory, df_seeder_factory, n_keys=2000
 
     await c_master.connection_pool.disconnect()
     await c_replica.connection_pool.disconnect()
+
+
+"""
+Test flushall command that's invoked while in full sync mode.
+This can cause an issue because it will be executed on each shard independently.
+More details in https://github.com/dragonflydb/dragonfly/issues/1231
+"""
+@pytest.mark.asyncio
+async def test_flushall_in_full_sync(df_local_factory, df_seeder_factory):
+    master = df_local_factory.create(port=BASE_PORT, proactor_threads=4, logtostdout=True)
+    replica = df_local_factory.create(port=BASE_PORT+1, proactor_threads=2, logtostdout=True)
+
+    # Start master
+    master.start()
+    c_master = aioredis.Redis(port=master.port)
+
+    # Fill master with test data
+    seeder = df_seeder_factory.create(port=master.port, keys=10_000, dbcount=1)
+    await seeder.run(target_deviation=0.1)
+
+    # Start replica
+    replica.start()
+    c_replica = aioredis.Redis(port=replica.port)
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+
+    full_sync_role = [b'replica', b'localhost', bytes(str(master.port), 'ascii'), b'full_sync']
+
+    # Wait for full sync to start
+    while await c_replica.execute_command("role") != full_sync_role:
+        await asyncio.sleep(0.0)
+
+    # Issue FLUSHALL and push some more entries
+    await c_master.execute_command("FLUSHALL")
+
+    connecting_role = [b'replica', b'localhost', bytes(str(master.port), 'ascii'), b'connecting']
+    while await c_replica.execute_command("role") == connecting_role:
+        await asyncio.sleep(0.0)
+
+    # Make sure we are still in full-sync mode
+    assert (await c_replica.execute_command("role") == full_sync_role
```

Comment:
why do we want this check in the test after the flush all?

Reply:
Because if the sync got to `stable_sync` too quickly, the test is meaningless (and we'll likely need to bump up the seeder config)

Reply:
but after the flush all the db is empty and you will most likely finish the stable sync very quickly. 
What you want to check in this test that flushall causes to start replication from start, i.e we get to the connecting state

Reply:
Is there a way to verify that? I imagine that connecting on the localhost takes very little time..

Reply:
I believe we can check that the sync id for the replica is not the same, as it is incremented every time a replica starts handshake

Reply:
Sure, done, but lmk if there's an elegant way to wait instead of the added `sleep()` there.

Reply:
I cant think of any other way other than sleep here